components/server/src/prometheusMetrics.ts

```diff
@@ -0,0 +1,19 @@
+import * as prometheusClient from 'prom-client';
+
+// Enable collection of default metrics.
+prometheusClient.collectDefaultMetrics({ timeout: 5000 });
+export const register = prometheusClient.register
+
+const loginCounter = new prometheusClient.Counter({
+    name: 'gitpod_login_requests_total',
+	help: 'Total amount of login requests',
+    labelNames: ['status', 'oauth'],
+    registers: [register]
+});
+
+export function increaseLoginCounter(status: string, oauth: string){
```

Comment:
now that we're here... can we add an additional label for the reason, please?

I understand, that a metric isn't meant to diagnose functional bug, but maybe it's helpful to decide on further hardening measures if we know the frequency of failed logins due to, e.g.
 - external service errors (e.g. code 500 from github.com)
 - browser history rewind to unrepeatable steps (e.g. swipe back on touchscreen/touchpad may lead to errors)
 - etc.

that would mean to introduce fine grained counter increments in:
 - https://github.com/gitpod-io/gitpod/blob/master/components/server/src/auth/login-completion-handler.ts
 - https://github.com/gitpod-io/gitpod/blob/master/components/server/src/auth/generic-auth-provider.ts
 - maybe more